[WebGPU] Bi-planar, non-packed 10-bit YCbCr videos are not correctly imported via importExternalTexture

mwyrzykowski opened the pull request:
#### 29b3635503789177104ab41d944ab2bfbb9c6f23
<pre>
[WebGPU] Bi-planar, non-packed 10-bit YCbCr videos are not correctly imported via importExternalTexture
<a href="https://bugs.webkit.org/show_bug.cgi?id=296456">https://bugs.webkit.org/show_bug.cgi?id=296456</a>
<a href="https://rdar.apple.com/156643840">rdar://156643840</a>

Reviewed by Dan Glastonbury.

Use the appropriate MTLPixelFormat for CVMetalTextureCache when importing
10-bit video formats.

Test: fast/webgpu/regression/repro_275345.html continues to pass

* Source/WebGPU/WebGPU/BindGroup.mm
* Source/WebGPU/WebGPU/MetalSPI.h:
Some 10-bit video formats were incorrectly handled during import.

Canonical link: <a href="https://commits.webkit.org/298022@main">https://commits.webkit.org/298022@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/66eac81955d20607d0f3844f85995d97ae5c1bc5

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/113376 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/33103 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/23565 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/119570 "Built successfully") | [💥 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/64333 "An unexpected error occured. Recent messages:Printed configuration") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/33739 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/41672 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/86284 "Passed tests") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/41368 "Unexpected infrastructure issue, retrying build") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/116323 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/26949 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/101962 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/66613 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/26214 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/63317 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/96329 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/20160 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/122799 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/40430 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/30183 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/95141 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/40818 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/98168 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/94887 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24276 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/40011 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/17828 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/36593 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/40315 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/45814 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/39967 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/43288 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/41711 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/6919a9a3d866c5cf4d4c5e07d1bd262533052ece. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48576

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/66eac81955d20607d0f3844f85995d97ae5c1bc5. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48576

djg reviewed (approved):
:shipit:

mwyrzykowski committed:
[WebGPU] Bi-planar, non-packed 10-bit YCbCr videos are not correctly imported via importExternalTexture
https://bugs.webkit.org/show_bug.cgi?id=296456
rdar://156643840

Reviewed by Dan Glastonbury.

Use the appropriate MTLPixelFormat for CVMetalTextureCache when importing
10-bit video formats.

Test: fast/webgpu/regression/repro_275345.html continues to pass

* Source/WebGPU/WebGPU/BindGroup.mm
* Source/WebGPU/WebGPU/MetalSPI.h:
Some 10-bit video formats were incorrectly handled during import.

Canonical link: https://commits.webkit.org/298022@main
